apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -96,7 +96,8 @@ verify(Data, Nonce, Evd, Target) when is_list(Evd) ->
 generate_hashed(_Hash, _Nonce, _Target, _Trims, _Threads, 0) ->
     {error, generation_count_exhausted};
 generate_hashed(Hash, Nonce, Target, Trims, Threads, Retries) when Retries > 0 ->
-    case generate_single(Hash, Nonce, Trims, Threads) of
+    Nonce32 = Nonce band 16#7fffffff,
```

Comment:
I see this limits the nonce to non-negative signed 32 bits integer.

@akorosmezey questions:
1. When a solution is found, `Nonce` is returned (line 106). That means the nonce returned is not the nonce used for computation. Is this not wrong?
2. I see the overflow to more than 32 bits is caused by the `Nonce + 1` in the recursion at line 108. Why not addressing this overflow in that line? (I assume that the initial nonce returned by `aec_pow:pick_nonce()` already complies to that - you could add guard in generate_hashed function for ensuring that is the case.)
3. Why non-negative *signed* 32 bits integer (`16#7fffffff`) rather than non-negative *unsigned* 32 bits integer (`16#ffffffff`)?